src/dbnode/encoding/proto/encoder.go

```diff
@@ -72,16 +72,32 @@ type Encoder struct {
 	varIntBuf              [8]byte
 	changedValues          []int32
 	fieldsChangedToDefault []int32
+	marshalBuf             []byte
 
 	unmarshaled *dynamic.Message
 
-	hardErr                          error
-	hasEncodedFirstSetOfCustomValues bool
-	closed                           bool
+	hardErr          error
+	hasEncodedSchema bool
+	closed           bool
 
+	stats            encoderStats
 	timestampEncoder m3tsz.TimestampEncoder
 }
 
+// EncoderStats contains statistics about the encoders compression performance.
+type EncoderStats struct {
+	UncompressedBytes int
+	CompressedBytes   int
+}
+
+type encoderStats struct {
```

Comment:
No because one of the fields is generated from another piece of state, so if I kept the public one on the encoder I'd end up wasting another 8 bytes per encoder (I know that seems like an over optimization but I have millions of these things in memory and they're already using 10% of memory not including the data so I'm trying to cut as much state out of them as possible to reduce mem usage)

Reply:
ah. yes, missed that on the first read. makes sense